File: src/main/java/controller/MainController.java
package ru.javarush.cryptoanalyzer.konovalov.controller;

import ru.javarush.cryptoanalyzer.konovalov.Commands.Action;
import ru.javarush.cryptoanalyzer.konovalov.entity.*;
import ru.javarush.cryptoanalyzer.konovalov.exception.AppException;

public class MainController {

    public Result doAction(String actionName, String[] parameters) {

        Action action = ActionContainer.getAction(actionName);

        try {
            return action.execute(parameters);
        }
        catch(AppException e){
            return new Result(ResultCode.ERROR, e.getMessage());
        }
    }
}

File: src/main/kotlin/controller/MainController.kt
package controller

import entity.ResultCode
import entity.Result
import exception.AppException

class MainController {
    fun doAction(actionName: String, parameters: Array<String>) : Result {
        val action = ActionContainer.getAction(actionName)

        try{
            return action.execute(parameters)
        }
        catch (e: AppException){
            return Result(ResultCode.ERROR, e.getMessaage())
        }
    }
}